crates/primitives/convert/src/felt.rs

```diff
@@ -48,7 +48,7 @@ mod tests {
         assert_eq!(felt_to_u128(&Felt::THREE).unwrap(), 3);
         assert_eq!(felt_to_u128(&Felt::from(u64::MAX)).unwrap(), u64::MAX as u128);
         assert_eq!(felt_to_u128(&Felt::from(u128::MAX)).unwrap(), u128::MAX);
-        assert!(felt_to_u128(&(Felt::from(u128::MAX) + Felt::ONE)).is_err());
-        assert!(felt_to_u128(&Felt::MAX).is_err());
+        assert!(matches!(felt_to_u128(&(Felt::from(u128::MAX) + Felt::ONE)), Err(FeltToU128Error)));
+        assert!(matches!(felt_to_u128(&Felt::MAX), Err(FeltToU128Error)));
```

Comment:
we can use `assert_matches` here

Reply:
👍

Reply:
you’re right that relying on the current state of the mainnet and the gateway isn’t ideal. However, this is a temporary solution. Storing the entire class within the project would be too heavy. For testing purposes, we need simple and lightweight class definitions from different versions. 

we can also open an issue to create dedicated test data (blocks, classes, etc.) that we can use in the future to further improve the reliability of our tests.